jwt.go

```diff
@@ -150,8 +150,11 @@ type JWTAuth struct {
 	logger        *zap.Logger
 	parsedSignKey interface{} // can be []byte, *rsa.PublicKey, *ecdsa.PublicKey, etc.
 
-	jwkCache     *jwk.Cache
-	jwkCachedSet jwk.Set
+	// JWK cache by resolved URL to support placeholders in JWKURL
+	jwkCaches map[string]*struct {
+		cache     *jwk.Cache
+		cachedSet jwk.Set
```

Comment:
Also, shall we create a struct type for the caches? I saw it has been used multiple times in the code later.